### PR TITLE
chore: Remove marketing consent and related code - WPB-11190

### PR DESCRIPTION
**Pull request by samwyndham**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11190" title="WPB-11190" target="_blank">TaskWPB-11190</a>  [iOS] Remove marketing consent functionality
  </td></table>
  <br />
 

### Issue

This issue removes marketing consent and related code which is currently buggy. 

If this PR is ever reverted please take a look at the comments of [[iOS] Marketing consent asked if app is disconnected from internet](https://wearezeta.atlassian.net/browse/WPB-10732) which discusses a selection of marketing consent bugs and possible fixes.

#### Scenario 1 - Creating account

1. From signed out account, select `Create a wire personal account`.
2. Go through the complete account creation process

##### Expectation

- Account creation goes smoothly
- There should be no forms asking for marketing consent / newsletters

#### Scenario 2 - Signing in through existing account

1. From signed out account, select `Login`.
2. Go through the login process with an existing account.

##### Expectation

- Signing in goes smoothly
- There should be no forms asking for marketing consent / newsletters

#### Scenario 3 - Settings

1. Sign in with an existing account
2. Go to Account > Settings > Account > Data Usage Permissions

##### Expectation

- There should be no settings related to marketing consent / newsletters

### Checklist

- [X] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [X] Description is filled and free of optional paragraphs.
- [X] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios/Wire-iOS/Resources/ar.lproj/Localizable.strings [hunk 1]
```diff
@@ -1917,13 +1917,6 @@
 "url_action.invalid_link.title" = "رابط غير صالح.";
 "url_action.invalid_link.message" = "إنّ الرابط الدي قمتَ بفتحه عير صالح.";
 
-// +++++++++++++++++++++++++++++++++++++++++++++++++++++
-// News and offers consent prompt
-
-"news_offers.consent.title" = "Do you want to receive news and product updates from Wire via email?";
```
Comment (netbe) [inline wire-ios/Wire-iOS/Resources/ar.lproj/Localizable.strings:1923]: are we not missing other languages? I have 20 results locally

**Review by netbe — COMMENTED** [review]
Left one comment before approving, otherwise looks good

File: wire-ios/Wire-iOS/Resources/ar.lproj/Localizable.strings (hunk 1 above)
Comment (samwyndham) [reply wire-ios/Wire-iOS/Resources/ar.lproj/Localizable.strings:1923]: Yes, you are right. I'm going to try an experiment with this PR to see if it they are automatically removed in the crowdin localization PR. If not I will remove them in a followup PR

**Review by KaterinaWire — APPROVED** [review]

File: wire-ios/Wire-iOS/Resources/ar.lproj/Localizable.strings (hunk 1 above)
Comment (netbe) [reply wire-ios/Wire-iOS/Resources/ar.lproj/Localizable.strings:1923]: yep, it should be removed by crowdin action, no pb.

**Review by netbe — APPROVED** [review]
looks good!

**Pull request merged**